### PR TITLE
json trace for online processing

### PR DESCRIPTION
**Pull request by vyzo**

### CODE REVIEW

File: traced/traced.go [hunk 1]
```diff
@@ -218,6 +228,47 @@ func (tc *TraceCollector) collectWorker() {
...
+	}
+	defer in.Close()
+
+	gzipR, err := gzip.NewReader(in)
+	if err != nil {
+		panic(err)
+	}
+	defer gzipR.Close()
+
+	out, err := os.OpenFile(tc.jsonTrace, os.O_WRONLY|os.O_TRUNC, 0644)
```
Comment (raulk) [inline traced/traced.go]: Hum, we have no guarantee that filebeat (or whoever is processing the json file) will get to it before it's replaced. Can we instead make `jsonTrace` a directory, and spit out a new file with the same base name as the binary .tar.gz, but replacing its extension with .json?

Filebeat will then erase the file when it's been successfully processed.

Comment (vyzo) [reply traced/traced.go]: sure, we can do that

Comment (vyzo) [reply traced/traced.go]: done.

File: traced/traced.go [hunk 2]
```diff
@@ -218,6 +229,47 @@ func (tc *TraceCollector) collectWorker() {
...
+
+	out, err := os.OpenFile(fmt.Sprintf("%s/%s.json", tc.jsonTrace, name), os.O_WRONLY|os.O_CREATE, 0644)
+	if err != nil {
+		panic(err)
+	}
+	defer out.Close()
+
+	var evt pb.TraceEvent
+	pbr := ggio.NewDelimitedReader(gzipR, 1<<20)
+	enc := json.NewEncoder(out)
```
Comment (raulk) [inline traced/traced.go:254]: We need to use https://godoc.org/google.golang.org/protobuf/encoding/protojson

>  This package produces a different output than the standard "encoding/json" package, which does not operate correctly on protocol buffer messages.

Comment (vyzo) [reply traced/traced.go:254]: follow up from trying to use that -- turns out it doesn't work with proto2 protobufs, so we'll have to punt on this.

**Review by raulk — APPROVED** [review]

**Pull request merged**